### DISCUSSION TITLE
Switch off creation of recovery files

### DISCUSSION THREAD

**Discussion by Miso-1** [MPh-py/MPh#64] [General]
Hi,

Is it possible to switch off the creation of recovery files when running a simulation? Turning of the generation inside COMSOL doesn't affect the generation when running COMSOL using Python.
My problem is that I have a huge amount of simulations to run parallel and the disk space is not enough for multiple recovery files.

Thanks and best regards
Michael

**Comment by john-hen** [comment]
Believe me, I've tried. Because I had the same problem. MPh already tries to enforce that, regardless of user settings:

https://github.com/MPh-py/MPh/blob/6bbcc8dafcfca3587f1f7b940629dd9458ba330f/mph/client.py#L206-L220

I'm not sure if that actually does anything. That is, I'm not sure if it effectively overrides the user setting in Comsol, i.e. whatever is in `comsol.prefs` or possibly `comsolserver.prefs`. Which is why I have also set `tempfiles.recovery.autosave=off` in both of these files. Only then, I think, does `client.java.useRecovery()` return `False`. Which should mean: no recovery files. And yet it *still* produces recovery files. Maybe less of them? I don't know. I have no idea what's up with that, like what Comsol does there.

The same happens with the GUI, by the way. Even though, like I said, I've turned off recovery files. But when I load a file in the GUI and save it, stuff appears in the `recoveries` folder inside `.comsol`. More stuff is created if I also solve the model. So, there's that. I don't think it has anything to do with how we call the Comsol API from Python. Though I could be missing something. Obviously we *notice* it more when running *many* simulations via Python.
